### PR TITLE
tests: net: socket: tcp: move to new ztest API

### PR DESCRIPTION
**Pull request by hunterhu3000**
Move net/socket/tcp tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

### CODE REVIEW

**Comment by hunterhu3000** [comment]
CI checks more than "twister -T tests/net/socket/tcp"
Will analyze this problem later

File: tests/net/tcp/src/main.c [hunk 1]
```diff
@@ -1484,23 +1484,4 @@ static void test_server_timeout_out_of_order_data(void)
 	net_tcp_put(ooo_ctx);
 }
 
-/** Test case main entry */
-void test_main(void)
-{
-	ztest_test_suite(test_tcp_fn,
-			 ztest_unit_test(test_presetup),
```
Comment (enjiamai) [inline tests/net/tcp/src/main.c:1491]: Try to put the test_presetup in the setup function? It seems like this has to be the first one.

**Comment by hunterhu3000** [comment]
This test case cannot be run with CONFIG_ZTEST_SHUFFLE =y, as some of the test case should be run with specific order.
Thanks to review.

**Review by rlubos — COMMENTED** [review]
Please rebase (conflict).

**Comment by rlubos** [comment]
@hunterhu3000 Something must've gone wrong during the rebase, the `*.orig` files shouldn't be here.

**Comment by hunterhu3000** [comment]
> Please rebase (conflict).

Rebased. Thanks to review

**Comment by hunterhu3000** [comment]
test_z_close_obstructed has to be run at last. If not
```
	zassert_equal(dropped_packets_before, dropped_packets_after,
		      "packets after close");
```
will fail. There will be 1 packet dropped during the test. The reason is unknown yet

**Review by enjiamai — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Review by enjiamai — APPROVED** [review]
It looks quite good for me.

**Review by rlubos — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**